### ISSUE TITLE
Contain code that will be rejected by a future version of Rust

### ISSUE THREAD

**Issue by psionic12** [simnalamburt/obj-rs#61]
When I build my project which dependents on this crate, the compiler reports that:
```
warning: the following packages contain code that will be rejected by a future version of Rust: obj-rs v0.7.0
note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 2`
```

**Comment by psionic12** [comment]
Here's the detail:
```
The following warnings were discovered during the build. These warnings are an
indication that the packages contain code that will become an error in a
future release of Rust. These warnings typically cover changes to close
soundness problems, unintended or undocumented behavior, or critical problems
that cannot be fixed in a backwards-compatible fashion, and are not expected
to be in wide use.

Each warning should contain a link for more information on what the warning
means and how to resolve it.


To solve this problem, you can try the following approaches:


- If the issue is not solved by updating the dependencies, a fix has to be
implemented by those dependencies. You can help with that by notifying the
maintainers of this problem (e.g. by creating a bug report) or by proposing a
fix to the maintainers (e.g. by creating a pull request):

  - obj-rs@0.7.0
  - Repository: https://github.com/simnalamburt/obj-rs
  - Detailed warning command: `cargo report future-incompatibilities --id 2 --package obj-rs@0.7.0`

- If waiting for an upstream fix is not an option, you can use the `[patch]`
section in `Cargo.toml` to use your own version of the dependency. For more
information, see:
https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section
        
The package `obj-rs v0.7.0` currently triggers the following future incompatibility lints:
> warning: trailing semicolon in macro used in expression position
>    --> /Users/x/.cargo/registry/src/index.crates.io-6f17d22bba15001f/obj-rs-0.7.0/src/error.rs:134:12
>     |
> 134 |         )));
>     |            ^
>     |
>    ::: /Users/x/.cargo/registry/src/index.crates.io-6f17d22bba15001f/obj-rs-0.7.0/src/raw/material.rs:30:26
>     |
> 30  |                     _ => make_error!(WrongNumberOfArguments, "Expected exactly 1 argument"),
>     |                          ------------------------------------------------------------------ in this macro invocation
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #79813 <https://github.com/rust-lang/rust/issues/79813>
>     = note: `#[allow(semicolon_in_expressions_from_macros)]` on by default
>     = note: this warning originates in the macro `make_error` (in Nightly builds, run with -Z macro-backtrace for more info)
> 

```

**Comment by Davidster** [comment]
weirdly I tried pointing to a local copy of obj-rs and it stopped giving the error. does the incompat report not run for crates that are reference with `path = "./xyz"`?

**Comment by simnalamburt** [comment]
I'm afraid there is no trailing semicolon at the end of macro. Is there any sample repository to reproduce this error?

https://github.com/simnalamburt/obj-rs/blob/8a5f690d974ea6994793722c398004cab5cbf81f/obj-rs/src/error.rs#L129-L136

**Comment by simnalamburt** [comment]
Automatically fixed by republishing the crate with newer rustc, Fixed by v0.7.1 release.

**Comment by Davidster** [comment]
Thank you for the fix!
